Flaky test with exception: access violation writing on Windows

In `test_plot_split_value_histogram` sometimes `WindowsError: exception: access violation writing` occurs
https://github.com/microsoft/LightGBM/blob/164818b029b02253796c9b2fe6391c572608f345/tests/python_package_test/test_plotting.py#L63

For the moment of creating this issue I saw it two times: in #3380 and #3397.

```
================================== FAILURES ===================================
__________________ TestBasic.test_plot_split_value_histogram __________________

self = <test_plotting.TestBasic testMethod=test_plot_split_value_histogram>

    @unittest.skipIf(not MATPLOTLIB_INSTALLED, 'matplotlib is not installed')
    def test_plot_split_value_histogram(self):
>       gbm0 = lgb.train(self.params, self.train_data, num_boost_round=10)

..\tests\python_package_test\test_plotting.py:64: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Miniconda\envs\test-env\lib\site-packages\lightgbm\engine.py:231: in train
    booster = Booster(params=params, train_set=train_set)
C:\Miniconda\envs\test-env\lib\site-packages\lightgbm\basic.py:2010: in __init__
    self.__get_eval_info()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <lightgbm.basic.Booster object at 0x000000000A248B88>

    def __get_eval_info(self):
        """Get inner evaluation count and names."""
        if self.__need_reload_eval_info:
            self.__need_reload_eval_info = False
            out_num_eval = ctypes.c_int(0)
            # Get num of inner evals
            _safe_call(_LIB.LGBM_BoosterGetEvalCounts(
                self.handle,
                ctypes.byref(out_num_eval)))
            self.__num_inner_eval = out_num_eval.value
            if self.__num_inner_eval > 0:
                # Get name of evals
                tmp_out_len = ctypes.c_int(0)
                reserved_string_buffer_size = 255
                required_string_buffer_size = ctypes.c_size_t(0)
                string_buffers = [
                    ctypes.create_string_buffer(reserved_string_buffer_size) for i in range_(self.__num_inner_eval)
                ]
                ptr_string_buffers = (ctypes.c_char_p * self.__num_inner_eval)(*map(ctypes.addressof, string_buffers))
                _safe_call(_LIB.LGBM_BoosterGetEvalNames(
                    self.handle,
                    self.__num_inner_eval,
                    ctypes.byref(tmp_out_len),
                    reserved_string_buffer_size,
                    ctypes.byref(required_string_buffer_size),
>                   ptr_string_buffers))
E               WindowsError: exception: access violation writing 0x00007FF9085F04DE

C:\Miniconda\envs\test-env\lib\site-packages\lightgbm\basic.py:3182: WindowsError
```

Both times it was in the following Windows job
https://github.com/microsoft/LightGBM/blob/164818b029b02253796c9b2fe6391c572608f345/.vsts-ci.yml#L119-L121

--- a/python-package/lightgbm/basic.py
+++ b/python-package/lightgbm/basic.py
@@ -2976,9 +2976,9 @@ class Booster(object):
         ptr_string_buffers = (ctypes.c_char_p * num_feature)(*map(ctypes.addressof, string_buffers))
         _safe_call(_LIB.LGBM_BoosterGetFeatureNames(
             self.handle,
-            num_feature,
+            ctypes.c_int(num_feature),
             ctypes.byref(tmp_out_len),
-            reserved_string_buffer_size,
+            ctypes.c_size_t(reserved_string_buffer_size),
             ctypes.byref(required_string_buffer_size),
             ptr_string_buffers))
         if num_feature != tmp_out_len.value:
@@ -3175,9 +3175,9 @@ class Booster(object):
                 ptr_string_buffers = (ctypes.c_char_p * self.__num_inner_eval)(*map(ctypes.addressof, string_buffers))
                 _safe_call(_LIB.LGBM_BoosterGetEvalNames(
                     self.handle,
-                    self.__num_inner_eval,
+                    ctypes.c_int(self.__num_inner_eval),
                     ctypes.byref(tmp_out_len),
-                    reserved_string_buffer_size,
+                    ctypes.c_size_t(reserved_string_buffer_size),
                     ctypes.byref(required_string_buffer_size),
                     ptr_string_buffers))
                 if self.__num_inner_eval != tmp_out_len.value:

